aw-client-rust/src/lib.rs

```diff
@@ -29,13 +29,11 @@ impl std::fmt::Debug for AwClient {
 }
 
 impl AwClient {
-    pub fn new(ip: &str, port: &str, name: &str) -> AwClient {
-        let baseurl = format!("http://{ip}:{port}");
+    pub fn new(baseurl: reqwest::Url, name: &str, hostname: String) -> AwClient {
         let client = reqwest::Client::builder()
             .timeout(std::time::Duration::from_secs(120))
             .build()
             .unwrap();
-        let hostname = gethostname::gethostname().into_string().unwrap();
```

Comment:
This unwrap is extremely unlikely, you would need to essentially have invalid utf-8 in your hostname to fail here.
Keep this functionality, but maybe replace the "unwrap" with an "expect" instead.

Reply:
I agree, very unlikely, but in that unlikely event, panic-ing doesn't make sense.  Maybe I should just strip out UTF-8 characters instead?

Reply:
Sounds good!